modules/utxo-lib/test/bitgo/psbt/Psbt.ts

```diff
@@ -40,6 +49,48 @@ function getScriptTypes2Of3() {
   return outputScripts.scriptTypes2Of3.filter((scriptType) => scriptType !== 'p2trMusig2');
 }
 
+describe('getSignatureCount', function () {
+  it('tx', function () {
+    const inputs = txnInputScriptTypes.map((scriptType) => ({ scriptType, value: BigInt(1000) }));
+    const outputs = txnOutputScriptTypes.map((scriptType) => ({ scriptType, value: BigInt(900) }));
+
+    (['unsigned', 'halfsigned', 'fullsigned'] as const).forEach((sign, signatureCount) => {
+      const txb = constructTxnBuilder(inputs, outputs, network, rootWalletKeys, sign);
+      const tx = sign === 'fullsigned' ? txb.build() : txb.buildIncomplete();
+      assert.strictEqual(getSignatureCount(tx), signatureCount);
+      tx.ins.forEach((input, inputIndex) => {
+        assert.strictEqual(
+          getSignatureCount([tx.ins[inputIndex]], 0),
+          inputIndex === 4 && signatureCount > 0 ? 1 : signatureCount
```

Comment:
This weirdness is because the taprootMusig2 (or is this a p2sh-pk?) only has 1 signature when fullsigned?

Reply:
p2sh-pk. Comment added.